webxrlayers-1.bs

```diff
@@ -0,0 +1,730 @@
+    1. Initialize |layer|'s [=XRProjectionLayer/session=] to |session|.
+    1. Initialize |layer|'s {{XRProjectionLayer/ignoreDepthValues}} as follows:
+        <dl class="switch">
+        <dt> If |layerInit|'s {{XRProjectionLayerInit/depth}} value is <code>false</code> and the [=XR Compositor=] will make use of depth values
+        <dd> Initialize |layer|'s {{XRProjectionLayer/ignoreDepthValues}} to <code>false</code>
+        <dt> Otherwise
+        <dd> Initialize |layer|'s {{XRProjectionLayer/ignoreDepthValues}} to <code>true</code>
+        </dl>
+    1. Initialize |layer|'s {{XRLayer/colorTextures}} as follows:
+         <dl class="switch">
+         <dt> If [=XRProjectionLayer/context=] is a {{WebGL2RenderingContext}} and textureTarget is TEXTURE_2D_ARRAY
+         <dd> Initialize |layer|'s {{XRLayer/colorTextures}} to a |array| with a single new instance of {{WebGLTexture}} created as a texture array with |context| and |layerInit|'s {{XRProjectionLayerInit/alpha}} values.
+         <dt> Else if textureTarget is TEXTURE_2D
+         <dd> Initialize |layer|'s {{XRLayer/colorTextures}} to a |array| with 2 new instances of {{WebGLTexture}} created as 2D texture with |context| and |layerInit|'s {{XRProjectionLayerInit/alpha}} values.
+         <dt> Otherwise
+         <dd> throw an {{InvalidStateError}} and abort these steps.
+         <dd> 
+         </dl>
+    1. Initialize |layer|'s {{XRLayer/depthStencilTextures}} as follows:
+        <dl class="switch">
+        <dt> If |layerInit|'s {{XRProjectionLayerInit/depth}} and {{XRProjectionLayerInit/stencil}} are not set
+        <dd> Initialize |layer|'s {{XRLayer/depthStencilTextures}} to an empty |array|
+        <dt> Else if [=XRProjectionLayer/context=] value is a {{WebGL2RenderingContext}} and textureTarget is TEXTURE_2D_ARRAY
+        <dd> Initialize |layer|'s {{XRLayer/depthStencilTextures}} to an |array| with a single new instance of {{WebGLTexture}} created as a texture array with |context| and |layerInit|'s {{XRProjectionLayerInit/depth}} and {{XRProjectionLayerInit/stencil}} values.
+        <dt> Else if textureTarget is TEXTURE_2D
+        <dd> Initialize |layer|'s {{XRLayer/depthStencilTextures}} to an |array| with 2 new instances of {{WebGLTexture}} created as 2D texture with |context| and |layerInit|'s {{XRProjectionLayerInit/depth}} and {{XRProjectionLayerInit/stencil}} values.
+        <dt> Otherwise
+        <dd> throw an {{InvalidStateError}} and abort these steps.
+        </dl>
+    1. Allocate and initialize resources compatible with |session|'s [=XRSession/XR device=], including GPU accessible memory buffers, as required to support the compositing of |layer|.
+    1. If |layer|’s resources were unable to be created for any reason, throw an {{OperationError}} and abort these steps.
+    1. [=/Resolve=] |promise| with |layer|.
+  1. Return |promise|.
+
+</div>
+
+ISSUE: define how scaleFactor works
+
+ISSUE: define "opaque" {{WebGLTexture}}
```

Comment:
We don't have any concept of opaque textures here, do we?

Reply:
no, not yet.